sdk/src/test/java/com/google/cloud/dataflow/sdk/transforms/display/DisplayDataTest.java

```diff
@@ -0,0 +1,558 @@
+/*
+ * Copyright (C) 2015 Google Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+package com.google.cloud.dataflow.sdk.transforms.display;
+
+import static com.google.cloud.dataflow.sdk.transforms.display.DisplayDataMatchers.hasDisplayItem;
+import static com.google.cloud.dataflow.sdk.transforms.display.DisplayDataMatchers.hasKey;
+import static org.hamcrest.Matchers.allOf;
+import static org.hamcrest.Matchers.empty;
+import static org.hamcrest.Matchers.everyItem;
+import static org.hamcrest.Matchers.hasItem;
+import static org.hamcrest.Matchers.hasKey;
+import static org.hamcrest.Matchers.hasSize;
+import static org.hamcrest.Matchers.is;
+import static org.hamcrest.Matchers.isEmptyOrNullString;
+import static org.hamcrest.Matchers.not;
+import static org.hamcrest.Matchers.notNullValue;
+import static org.hamcrest.Matchers.nullValue;
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertThat;
+
+import com.google.cloud.dataflow.sdk.transforms.PTransform;
+import com.google.cloud.dataflow.sdk.transforms.display.DisplayData.Builder;
+import com.google.cloud.dataflow.sdk.transforms.display.DisplayData.Item;
+import com.google.cloud.dataflow.sdk.values.PCollection;
+import com.google.common.testing.EqualsTester;
+
+import org.hamcrest.FeatureMatcher;
+import org.hamcrest.Matcher;
+import org.hamcrest.Matchers;
+import org.joda.time.Duration;
+import org.joda.time.Instant;
+import org.junit.Rule;
+import org.junit.Test;
+import org.junit.rules.ExpectedException;
+import org.junit.runner.RunWith;
+import org.junit.runners.JUnit4;
+
+import java.util.Collection;
+import java.util.Map;
+
+/**
+ * Tests for {@link DisplayData} class.
+ */
+@RunWith(JUnit4.class)
+public class DisplayDataTest {
+  @Rule public ExpectedException thrown = ExpectedException.none();
+
+  @Test
+  public void testTypicalUsage() {
+    final HasDisplayData subComponent1 =
+        new HasDisplayData() {
+          @Override
+          public void populateDisplayData(DisplayData.Builder builder) {
+            builder.add("ExpectedAnswer", 42);
+          }
+        };
+
+    final HasDisplayData subComponent2 =
+        new HasDisplayData() {
+          @Override
+          public void populateDisplayData(DisplayData.Builder builder) {
+            builder.add("Location", "Seattle").add("Forecast", "Rain");
+          }
+        };
+
+    PTransform<?, ?> transform =
```

Comment:
I'm not sure I understand what you're proposing. Are you suggesting creating some sort of "spy builder" in order to call `populateDisplayData()` directly and verify interactions? Could be useful for the `ParDo` and `PTransfrom` tests, although I don't think it's appropriate here since we're really testing how the builder works.

Also, this particular test (`testTypicalUsage()`) is meant as a sort of integration test and example to show the intended usage of the interface.